src/components/opoTable/opoTable.js

```diff
@@ -30,72 +42,125 @@ export default function OpoTable({
   );
+  const createCol = accessor => {
+    let col = {
+      Header: (
+        <Heading
+          title={accessor}
+          color={accessor === "shadow" ? "red" : null}
+        />
+      ),
+      accessor: accessor,
+    };
+    if (accessor === "donors" || accessor === "investigation")
+      col.cellClass = "text-center";
+    if (accessor === "shadow") {
+      col.cellClass = styles.shadows
+      col.color = "red"
+    };
+    if (accessor === "tier") {
+      col.Cell = props => (
+        <Container>
+          <Tier className={styles.tierCol} tier={props.value} />
+        </Container>
+      );
+      col.cellClass = styles.tierCol;
+    }
+    return col;
+  };
+  return cols
+    .map(col => {
+      if (inState && col === "states") return col;
+      if (!inState && (col === "region" || col === "investigation")) return col;
+      return createCol(col);
+    })
+    .filter(head => typeof head !== "string");
+}, [headings, inState, citations]);
 
-  return (
-    <Row className={styles.opoTable}>
-      <h3>{title}</h3>
-      <Table striped>
-        <thead>
-          <tr>
-            {inState ? (
-              <>
-                <Heading title="name" />
-                <Heading title="region" />
-              </>
-            ) : (
-              <>
-                <Heading title="states" />
-                <Heading title="name" />
-              </>
-            )}
-            <Heading title="tier" />
-            <Heading title="donors" />
-            <Heading title="shadow" color="red" />
-            {inState && <Heading title="investigation" />}
-          </tr>
-        </thead>
-        <tbody>
-          {opos.map(
-            ({
-              donors,
-              investigation,
-              name,
-              region,
-              shadows,
-              states,
-              tier,
-            }) => (
-              <tr key={name}>
-                {inState ? (
-                  <>
-                    <td>{name}</td>
-                    <td>{region}</td>
-                  </>
-                ) : (
-                  <>
-                    <td>{states}</td>
-                    <td>{name}</td>
-                  </>
-                )}
-                <td>
-                  <Container>
-                    <Tier className={styles.tierCol} tier={tier} />
-                  </Container>
-                </td>
-                <td className="text-center">{formatNumber(donors)}</td>
-                <td className={styles.shadows}>{formatNumber(shadows)}</td>
-                {inState && (
-                  <td className="text-center">
-                    {investigation ? "Yes" : "--"}
-                  </td>
-                )}
-              </tr>
-            )
-          )}
-        </tbody>
-      </Table>
-    </Row>
-  );
-}
+  const data = useMemo(() => {
+    const formatNumber = (num, options) =>
+      !num || isNaN(num) ? "--" : num.toLocaleString("en-US", options);
```

Comment:
Why not pull this from `util` like the old code?

Reply:
react-table requires useMemo. if you're bringing in a function from outside the useMemo, you need to list it in the dependencies array. Everytime that function is called, it will trigger useMemo to recalculate. We could turn that into a useCallback, but that can only be done within a react component. So, easiest way is to redefine it since a. it is simple. and b. it isn't likely to change.

Reply:
> Everytime that function is called, it will trigger useMemo to recalculate

I would think the dependency never materially changed so no re-render? Any idea why that's happening?

Reply:
It caused an infinite loop when I tried it 🤷 
Maybe the value is the same but it's pointing to a different location in memory?

Reply:
🤷🏻 

like you said, not really worth investigating now, but curious